### PR TITLE
Add spec for `Converter`

### PR DESCRIPTION
**Pull request by moozzi**
#1

### CODE REVIEW

File: spec/converter_spec.rb [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+require 'ronin/masscan/converter'
+require 'masscan/output_file'
+require 'tempfile'
+
+RSpec.describe Ronin::Masscan::Converter do
+  let(:fixtures_path) { File.expand_path(File.join(__dir__, '..', 'spec', 'fixtures')) }
+  let(:masscan_path)  { File.join(fixtures_path, 'converter', 'masscan.json') }
+  let(:masscan_file)  { Masscan::OutputFile.new(masscan_path) }
+  let(:expected_json) do
+    "[{\"status\":\"open\",\"protocol\":\"tcp\",\"port\":80,\"reason\":[\"syn\",\"ack\"],\"ttl\":54,\"ip\":\"93.184.216.34\",\"timestamp\":\"2021-08-26 08:50:21 +0000\"}]"
```
Comment (AI-Mozi) [inline spec/converter_spec.rb:11]: Should I compare whole strings, or parse it back to JSON and compare some keys?
The `timestamp` may be different based on your location.

File: spec/converter_spec.rb [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  end
+
+  describe '.convert_file' do
+    it 'must convert masscan file and wirte it into a file' do
```
Comment (AI-Mozi) [inline spec/converter_spec.rb:15]: Should I add more examples?

File: spec/converter_spec.rb [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        expect(output_file.read).to eq(expected_json)
+      end
+    end
+  end
+
+  describe '.convert' do
+    context 'when there is no output' do
+      context 'and format is csv' do
+        let(:expected_csv) do
+          "type,status.status,status.protocol,status.port,status.reason,status.ttl,status.ip,status.timestamp,banner.protocol,banner.port,banner.ip,banner.timestamp,banner.app_protocol,banner.payload\nstatus,open,tcp,80,\"syn,ack\",54,93.184.216.34,2021-08-26 08:50:21 +0200\n"
```
Comment (AI-Mozi) [inline spec/converter_spec.rb:29]: Is there a way to make it multi-line but without adding some unnecessary `\n`'s?

File: spec/converter_spec.rb (hunk 1 above)
Comment (postmodern) [reply spec/converter_spec.rb:11]: I would define the expected JSON as a Ruby object then call `.to_json` on it.

```ruby
let(:expected_json)
  [
    {...}
  ].to_json
end
```

File: spec/converter_spec.rb (hunk 3 above)
Comment (postmodern) [reply spec/converter_spec.rb:29]: Do you mean the last newline? You can do either:

```ruby
<<~CSV.chomp
  foo,bar,baz
  ...
CSV
```

or

```ruby
[
  "foo,bar,baz",
  "..."
].join("\n")
```

File: spec/converter_spec.rb (hunk 2 above)
Comment (postmodern) [reply spec/converter_spec.rb:15]: You could add specs for when `input_format:` is given, or when `format:` is explicitly given.

File: spec/converter_spec.rb (hunk 3 above)
Comment (AI-Mozi) [reply spec/converter_spec.rb:29]: Not the last one. I don't want this `expected_csv` to be too long but saving it as `<<CSV.chomp ...` fails when I want to compare it with result of `.convert`

File: spec/fixtures/converter/masscan.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+[
+  { "ip": "93.184.216.34", "timestamp": "1629960621", "ports": [ {"port": 80, "proto": "tcp", "status": "open", "reason": "syn-ack", "ttl": 54} ] }
```
Comment (postmodern) [inline spec/fixtures/converter/masscan.json:2]: Should rebase this branch to get the new `spec/fixtures/converter/input.json`.

File: spec/converter_spec.rb (hunk 2 above)
Comment (postmodern) [reply spec/converter_spec.rb:48]: If we had a `spec/fixtures/converters/input.csv` file, we could then test `input_format: :csv, format: :json`.

**Review by postmodern — CHANGES_REQUESTED** [review]
Should rebase the branch against `main` to get the new `spec/fixtures/converters/input.json` file. Also I think we need an `input.csv` file as well so we can test `input_format: :csv`.

**Comment by AI-Mozi** [comment]
> If we had a spec/fixtures/converters/input.csv file, we could then test input_format: :csv, format: :json.

I'll add it later.

**Review by postmodern — APPROVED** [review]

**Pull request merged**